### PR TITLE
AIRFLOW-5529 Add Apache Drill provider.

### PR DESCRIPTION
**Pull request by jnturton**
This PR adds a new provider to Airflow for executing SQL queries against an Apache Drill instance.  This is useful because, while its primary use case is interactive analysis that often avoids the need for ETL, Drill's CTAS statements and support for querying a large variety of data sources also make it a very capable ETL tool.  What is lacking to use Drill this way is worklow management, and that gap is one that Airflow fills perfectly.

### CODE REVIEW

**Comment by dzamo** [comment]
I see I need to get the v1.1.0 sqlalchemy-drill release pushed to PyPI to proceed.  I'll do that and trigger another build.

**Comment by potiuk** [comment]
> I see I need to get the v1.1.0 sqlalchemy-drill release pushed to PyPI to proceed. I'll do that and trigger another build.

Yep :). Glad to see another apache-family provider added :). I will take a look shortly. It looks cool and might be just-in-time for the next wave of providers (next week likely)

**Comment by potiuk** [comment]
Some static checks + docs failing. For static checks I recommend installing `pre-commit` - it should fix the problems automatically.

**Comment by dzamo** [comment]
Thanks for the help @potiuk.  I've applied pre-commit and added a bit to the docs while I was investigating the error there.  I fear I haven't yet actually resolved the docs problem, "doc not present in any toctree", but I'm still a little puzzled about what more I might need to add, and also by what gets generated and what is written by hand.  So I've pushed the new commits to see what the error messages look like now...

**Comment by potiuk** [comment]
Take a look now at the build docs errors /static checks  errors, I think they tell exactly what to do :)

**Comment by potiuk** [comment]
The two errors now are not obvious indeed:

```
------------------------------ Error   1 --------------------
/opt/airflow/docs/apache-airflow-providers-apache-drill/_api/drill/index.rst: WARNING: document isn't included in any toctree
```

Here, i think (@mik-laj ?) "drill" has to be added to this list:

https://github.com/apache/airflow/blob/d3f300fba8c252cac79a1654fddb91532f44c656/docs/conf.py#L537

```
------------------------------ Error   2 --------------------
/opt/airflow/docs/apache-airflow-providers-apache-drill/commits.rst: WARNING: document isn't included in any toctree
```

Not obvioius, I know, but copy this below the "AIUTOGENERATED" comment. I will improve it in the future when someone adds a new provider so that this is not misleading:  

```
.. toctree::
    :maxdepth: 1
    :caption: Commits

    Detailed list of commits <commits>
```

File: airflow/providers/apache/drill/hooks/drill.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#
```
Comment (mik-laj) [inline airflow/providers/apache/drill/hooks/drill.py:1]: Can you create `__init__.py` file in hooks directory?

Comment (jnturton) [reply airflow/providers/apache/drill/hooks/drill.py:1]: Ah, how did I miss that?  Thanks.

**Comment by dzamo** [comment]
@potiuk Thank you for your help, I think I've now got the relevant tests passing.  There is still a test failure under "MySQL5.7, Py3.6: Always API Core Other CLI Providers WWW Integration" but I don't think I've touched anything related to that with this PR...

**Comment by potiuk** [comment]
Transient error :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Wohoo!

**Comment by potiuk** [comment]
We have a new Apache Provider ! Release coming this week :)

**Comment by dzamo** [comment]
Excellent!

**Pull request merged**